src/custom-components/Pagination/usePagination.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect } from "react";
+
+export const DOTS = "...";
+
+interface usePaginationInterface {
+  totalCount: number;
+  pageSize: number;
+  currentPage: any;
+}
+
+const range = (start: number, end: number) => {
+  let length = end - start + 1;
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+var paginationRange: any[] = [];
```

Comment:
I saw that it is done 👍. Resolving the conversation...

Reply:
I understand that it would be less confusing for other developers.
Other options would be to use useMemo to avoid generating a state or a local variable since it allows a non-null return. But this is also confusing. Another option would also be to use useRef instead of a local var, but it's the same case as var local vs useState, which will be better?
Anyway, I think it's okay for us to use a useState to avoid this confusion.

Reply:
I think `useMemo` does not fit here because it is mutable, right?

Since you need to re-render the dependent components when `paginationRange` is updated, I think `useState` is the best option.

Reply:
we're not really interested in maintaining a state between renders, we just want to calculate the value from the received properties and display it between renders.
So we don't really need a persistence state. That's why with a local variable it works.

if you get your data from props, the component will be re-rendered on props change anyway, so the view will be in sync with the data. Putting them in a state would just cause unnecessary rendering, first on the prop change, then on the state change.

Regarding using useMemo, it seems to me that it is more performant since it causes fewer renderings, does not need to wait for the vdom, less code and more readable. It does not introduce unnecessary state. But with the big but of whoever comes to this code, it will be difficult for him to understand the why of the whys.

Reply:
I think that we can continue the discussion in https://github.com/FromDoppler/doppler-sms-webapp/pull/4/files#r1061630903